### PR TITLE
chore: data pipeline tests base setup

### PR DESCRIPTION
**Pull request by mrehan27**
helps: https://github.com/customerio/issues/issues/11965

###  Changes

- Setting up tests environment for `DataPipeline`
- Updated `DataPipeline` so it can work with fake instances
- Added support methods to make testing `DataPipeline` easier
- Changed the name from `CustomerIOImplementationTest` to `DataPipelineInteractionTests` so it is easier to track updated tests

_**Note:** This is the main PR where we'll add more test changes later on_

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -81,6 +81,9 @@ let package = Package(
         .target(name: "CioDataPipelines",
                 dependencies: ["CioInternalCommon", .product(name: "Segment", package: "Segment")],
                 path: "Sources/DataPipeline"),
+        .testTarget(name: "DataPipelineTests",
```
Comment (ami-aman) [inline Package.swift:84]: Wondering if we name it as `CioDataPipelineTests` to maintain consistency ? Just thinking out loud !

File: Tests/DataPipeline/Support/TestUtilities.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    func execute<T>(event: T?) -> T? where T: RawEvent {
+        lastEvent = event
+        var eventSummary = ""
+
+        if let event = event {
+            eventSummary += " Event type: \(event.type ?? "nil")"
+            events.append(event)
+            if let event = event as? TrackEvent {
```
Comment (Shahroz16) [inline Tests/DataPipeline/Support/TestUtilities.swift:25]: We just want to print events of type `TrackEvent`?

**Review by Shahroz16 — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (mrehan27) [reply Package.swift:84]: None of our test package has `Cio` prefix. It is only for main packages, probably to keep them unique for public apis. So no changes needed here.

File: Tests/DataPipeline/Support/TestUtilities.swift (hunk 2 above)
Comment (mrehan27) [reply Tests/DataPipeline/Support/TestUtilities.swift:25]: Just printed event name, which is available in `TrackEvent` only. For rest of the events e.g. `identify`, `type` is generally helpful.

**Review by ami-aman — APPROVED** [review]

**Pull request merged**